bazel/ray_deps_setup.bzl

```diff
@@ -1,128 +1,198 @@
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository", "new_git_repository")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-def ray_deps_setup():
-    RULES_JVM_EXTERNAL_TAG = "2.10"
-    RULES_JVM_EXTERNAL_SHA = "1bbf2e48d07686707dd85357e9a94da775e1dbd7c464272b3664283c9c716d26"
+def github_repository(**kwargs):
```

Comment:
I can if you really want to, but are do you really prefer having to update this function every time someone wants to use an argument that wasn't previously in the supported list? I tried to make this transparent so it would just pass unknown arguments through.

Reply:
Yes that would be preferred

Reply:
Ok I'll change it

Reply:
@edoakes Wait I just realized I can't do that, because if I put them here then people will try to call them positionally rather than as keywords. And Starlark (which is this Python-like language) forbids keyword-only arguments, so I can't do that either.

That said though, I'll try to shuffle the logic to make it easier to see what the defaults are.

Reply:
Update: see new commits.